segment/analytics/request.py

```diff
@@ -37,26 +44,32 @@ def post(write_key, host=None, gzip=False, timeout=15, proxies=None, **kwargs):
 
     kwargs = {
         "data": data,
-        "auth": auth,
         "headers": headers,
         "timeout": 15,
     }
 
     if proxies:
         kwargs['proxies'] = proxies
-
-    res = _session.post(url, data=data, auth=auth,
-                        headers=headers, timeout=timeout)
-
+    res = None
+    try:
+        res = _session.post(url, data=data, headers=headers, timeout=timeout)
+    except Exception as e:
+        log.error(e)
+        raise e
```

Comment:
why do you `log.error` and then `raise`? That is kind of redundant, I would like to know if there is a good reason for this

Reply:
A bit verbose maybe.  This could raise somewhere like a retry mechanism where it doesn't get logged.  If there's a problem with, please do raise an issue ticket